File: Servidor/Transaccional/Comandos/Comandos/ComandoBase.cs

namespace Trascend.Bolet.Comandos.Comandos
{
    /// <summary>
    /// Comando que devuelve un objeto genérico
    /// </summary>
    /// <typeparam name="T">Objeto genérico</typeparam>
    public abstract class ComandoBase<T>
    {
        Receptor<T> _receptor = null;

        /// <summary>
        /// Propiedad que asigna u obtiene el objeto genérico
        /// </summary>
        public Receptor<T> Receptor
        {
            get { return _receptor; }
            protected set { _receptor = value; }
        }

        /// <summary>
        /// Método que Ejecuta la acción del comando
        /// </summary>
        public abstract void Ejecutar();
    }
}

File: trunk/src/Servidor/Transaccional/Comandos_Fac/Comandos/ComandoBase.vb
Namespace Comandos
    ''' <summary>
    ''' Comando que devuelve un objeto genérico
    ''' </summary>
    ''' <typeparam name="T">Objeto genérico</typeparam>
    Public MustInherit Class ComandoBase(Of T)
        Private _receptor As Receptor(Of T) = Nothing

        ''' <summary>
        ''' Propiedad que asigna u obtiene el objeto genérico
        ''' </summary>
        Public Property Receptor() As Receptor(Of T)
            Get
                Return _receptor
            End Get
            Protected Set(ByVal value As Receptor(Of T))
                _receptor = value
            End Set
        End Property

        ''' <summary>
        ''' Método que Ejecuta la acción del comando
        ''' </summary>
        Public MustOverride Sub Ejecutar()
    End Class
End Namespace